[FRONTEND] Vendor page

abdullahyldz opened the pull request:
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other

# Description

Vendor's profile page, which is accessible to both customers and vendors.
VendorHomePage is consisted of two components VendorSplash and TabsPane that consists UserReviews and VendorMainContent. UserReviews will show the reviews of the vendor that customers make after they make an order. VendorMainContent will display all the products of the vendor.
VendorSplash shows some information about vendor like 'title', 'description', 'profile picture' and 'rating' on marble background picture.

Page is designed to be able to edited by its owner. No other person can edit it. First check for owner is at splash, which shows Edit Mode switch to owner and Send Message button to others. 
Also, product cards get another props called 'EditMode' to become to its editable version. Edit Mode in the splash is needed to be on to make products editable, also product cards check for the current user to be the owner of the product.


# Screenshots

In case your pull request does not consist of visual changes, you can delete this section. 
In customer and other vendor's eyes:
![image](https://user-images.githubusercontent.com/44057640/104843452-cc15e080-58db-11eb-9d59-e07b1a35fccb.png)
In vendor owner's eyes
![image](https://user-images.githubusercontent.com/44057640/104843544-434b7480-58dc-11eb-8c63-1cfe9bc315cc.png)
Some background, border, margin changes:
![image](https://user-images.githubusercontent.com/44057640/104843985-8a3a6980-58de-11eb-99d0-2b9eb5b8a693.png)

# Test results

Please run all the tests and dump the result/summary here in a code block.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have checked my code and corrected any misspellings
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# Potential problems/Important remarks

In the second image, you can see some products are not in edit mode because they are other vendor's products. Normally, they shouldn't be in this page. 
Also, I am signing in as Can Batuk which has id like 55 or something but the products who has vendor Can Batuk has id 3. So, I played in /init method to access edit mode. We should learn the credentials from backend.

/closes #358

abdullahyldz committed:
vendor homepage header design

abdullahyldz committed:
edit page redirection

abdullahyldz committed:
editable product card

abdullahyldz committed:
editable product card

abdullahyldz committed:
vendor page layout pane

abdullahyldz committed:
vendor reviews pane

abdullahyldz committed:
add to card button is disabled when vendor

abdullahyldz committed:
verified notif. when guest bug

abdullahyldz committed:
editable products

abdullahyldz committed:
vendor header edit page button

abdullahyldz committed:
shopp.cart icon removed for vendor

abdullahyldz committed:
vendor rate display centered

abdullahyldz committed:
replace edit profile with send message for nonvendors

abdullahyldz committed:
editing mode changed to vendor and owner

ozdenizdolu committed:
Create folder&file structure

ozdenizdolu committed:
create simple skeletal structure

ozdenizdolu committed:
Multiple changes
- Create a temporary route /productModal for testing
- Create a steps structure for product modal inner
- Add control buttons to steps structure

ozdenizdolu committed:
Merge branch 'master' into F-product-modal

abdullahyldz committed:
vendor page content layout

abdullahyldz committed:
file restructure vendor page

abdullahyldz committed:
vendor splash file split

abdullahyldz committed:
editable changes

abdullahyldz committed:
rating bug fit content

abdullahyldz committed:
background image on vendor splash

abdullahyldz committed:
rating color fix

abdullahyldz committed:
changes

abdullahyldz committed:
switch edit mode

abdullahyldz committed:
add product button in tabs pane

abdullahyldz committed:
remove login/init id changes for edit mode

abdullahyldz committed:
remove logs

abdullahyldz committed:
rmv logs

abdullahyldz committed:
add product button invisible to customers

abdullahyldz committed:
margins background border

MehdiSaffar commented on ecommerce/frontend/src/components/VendorContent/VendorSplash.js:
```diff
@@ -0,0 +1,61 @@
+                </div>
+                <div className="vendor-edit-button">
+                    {isVendorAndOwner ? (
+                        <div className="vendor-edit-mode-open">
+                            <p>{editable ? 'Edit Mode' : 'Seeing as Customer'}</p>
+                            <Switch defaultChecked={true} onChange={onEditModeChange} />
+                        </div>
```
Since you use `onChange` shouldnt you also assign `value={editable}`?

abdullahyldz replied:
Switch component handles its own state.

MehdiSaffar commented on ecommerce/frontend/src/components/VendorContent/VendorSplash.js:
```diff
@@ -0,0 +1,61 @@
+                            <Switch defaultChecked={true} onChange={onEditModeChange} />
+                        </div>
+                    ) : (
+                        <Button type="primary" icon={<EditOutlined />} href="/profile/messages">
+                            Send a message
+                        </Button>
```
Not sure if Button has href but in any case a regular html link shouldn't be used. You need to use either Link from react-router or have a onClick which uses `history.push('/profile/messages/`). 

Also messages page does not allow creating a new conversations. For this you will need to use a  "new message modal". We are working on it with @ozdenizdolu in F-prepare-vendor-ui branch.

abdullahyldz replied:
Ok, then I will use onClick function, which first opens a modal for 'Sending a Message', after sending a message function uses 
`history.push('/profile/messages/').` and finishes.

MehdiSaffar commented on ecommerce/frontend/src/components/layout/Content.js:
```diff
@@ -28,7 +29,7 @@ export const Content = () => {
     return (
         <Layout.Content className="content">
             <CategoryBar />
-            {location.pathname.startsWith('/profile') || <IsVerifiedNotification />}
+            {isUser && user.is_verified && <IsVerifiedNotification />}
 
```
the reason we have the `location.pathname` is because when you are in the profile age the notification will appear twice, both in Content and in Profile page. maybe you can add those `isUser...` conditions?

abdullahyldz replied:
I hadn't seen this component.
![image](https://user-images.githubusercontent.com/44057640/105630080-e01b8e00-5e57-11eb-9923-6870ce682a4d.png)

MehdiSaffar commented on ecommerce/frontend/src/components/layout/Header.js:
```diff
@@ -72,14 +72,16 @@ export const CustomerHeaderContent = () => {
-                </Button>
-            </Link>
+            {!isVendor && (
+                <Link to="/shoppingCart" className="header-customer-cart">
+                    <Button
+                        className="header-customer-cart"
+                        icon={<ShoppingCartOutlined className="header-customer-cart-icon" />}
+                        ghost>
+                        My Cart
+                    </Button>
+                </Link>
```
for that previous button i mentioned (the one with href), maybe you can use the same code here?

MehdiSaffar commented on ecommerce/frontend/src/components/pages/VendorHomepage.js:
```diff
@@ -0,0 +1,87 @@
+        setEditMode(checked)
+    }
+
+    const onAddProduct = () => {
+        console.log('Adding product here')
+    }
```
Please implement this (at least using mock) before merging the pr

MehdiSaffar commented on ecommerce/frontend/src/components/pages/VendorHomepage.js:
```diff
@@ -0,0 +1,87 @@
+
+export const VendorHomepage = props => {
+    const { vendorId } = props.match.params
+    const [editMode, setEditMode] = useState(true)
```
shouldn't default value of editMode be the value of `isVendorAndOwner`?

MehdiSaffar commented on ecommerce/frontend/src/components/pages/VendorHomepage.js:
```diff
@@ -0,0 +1,87 @@
+import { PlusCircleOutlined } from '@ant-design/icons'
+
+const { TabPane } = Tabs
+function callback(key) {
+    console.log(key)
+}
```
what is this used for?

abdullahyldz replied:
deleted

MehdiSaffar commented on ecommerce/frontend/src/components/product_card/ProductCard.js:
```diff
@@ -1,33 +1,65 @@
         addShoppingCartItem(product, 1)
     }
 
+    const onDeleteProductCard = () => {
+        console.log('delete product card')
+    }
+    const onEditProductCard = () => {
+        console.log('edit product card')
+    }
```
Please implement these in this PR

MehdiSaffar commented on ecommerce/frontend/src/components/product_card/ProductCard.js:
```diff
@@ -1,33 +1,65 @@
+    const { user } = useAppContext()
+    const isVendor = user.type === 'vendor'
+    const isVendorAndOwner = isVendor && vendor.id === user.id
+    let editableProduct = editMode && isVendorAndOwner
```
should be `const`

MehdiSaffar commented on ecommerce/frontend/src/components/product_card/ProductCard.js:
```diff
@@ -1,33 +1,65 @@
-    const { title, rating, price, price_after_discount, currency, images, id } = product
-
-    // const discountPrice = price * (1 - discount)
+    const { title, rating, price, price_after_discount, currency = '₺', images, id, vendor } = product
```
Can you make currency TL instead just like we used everywhere?

abdullahyldz replied:
sure

MehdiSaffar commented on ecommerce/frontend/src/components/VendorContent/VendorSplash.js:
```diff
@@ -0,0 +1,61 @@
+}
+
+export const VendorSplash = ({ vendorId, onEditModeChange, editable }) => {
+    const vendorHeaderDetails = {
+        title: 'SAMSUNG',
+        imageUrl:
+            'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAJcAAABlCAMAAAB3G4FZAAAAz1BMVEX///8LIIn+/v7///0LIYj///sAAIAAAHwAAIf///kAAIQMH4rg4u6ws8oAAHTFy9o4RJXL0dRyeKtiap8AG4gAF4kAAG/i6PEADIoLIoQAEYYADocAAIzx8vfe4enz9vft7uwsNZDm5+rBxc/Y2OUpNomxtsajqb6TlbSCg65dYJ1ISY4NIHs9R5CEh7eeorpET5AeKoK+wNFze6JUVJeip8Z3d6NmaJR/hqqRlb5XX5Nyc64eK4onNn/u8eaUmqmFiaM5PYBQUqI7QpxQVIvd7jXQAAAJNElEQVRoge2ZaWPauhKGBZJl ...
```
don't forget to remove this once done

abdullahyldz replied:
I added mock data. These are gone, now.

MehdiSaffar reviewed (changes requested):
I like the design change and I overall think the code is solid.

Please add some mock handlers in mock/handlers.js so that this can be tested with backend.

ozdenizdolu committed:
Product info form added in product modal

ozdenizdolu committed:
Merge branch 'master' into F-product-modal

ozdenizdolu committed:
Add image upload section to product modal

abdullahyldz committed:
send message on click history push

abdullahyldz committed:
refactor, callback del

abdullahyldz committed:
variable type change in product card

abdullahyldz committed:
bug fix

bekir96 committed:
[BACKEND] B-create-product-list (#402)

* create product_list endpoint started

* product_list_create is defined and product_list.py is added under views folder

* product_list_create is added to init of views folder

* Test for create product list is added and also getflix urls are arranged for new endpoint

* Response message typo is fixed

* duplicate name error in the list is fixed

Co-authored-by: omerfdeniz <omerfaruk.deniz@gmail.com>

* test for duplicate list name error is added

Co-authored-by: omerfdeniz <omerfaruk.deniz@gmail.com>

bekir96 committed:
[BACKEND] B-get-product-list (#404)

* product list get method implemented and remove unused field(name) from product table, also migration is satisfied

* Product list serializer is added and get product list method is arranged by it

* test for get product list is added

bcetin committed:
[B] Adding Vendor Details Endpoint (#390)

Adding vendor/details endpoint where one can get the details of a vendor.

bekir96 committed:
[BACKEND] B-delete-product-list (#406)

* Product list delete is added and list/{id} endpoint is added to urls.py

* test for delete product list is added

* when error check of no list with given id is satisfied, return message should be more informative

Co-authored-by: omerfdeniz <omerfaruk.deniz@gmail.com>

* Possible table migration fault on main is fixed

Co-authored-by: omerfdeniz <omerfaruk.deniz@gmail.com>

bekir96 committed:
[BACKEND] B-add-product-to-list (#409)

* Add product to list implementation started

* add product to list method implemented in product_list.py

* test for add product to list is added

* prevent duplicate product entries in list is added

Co-authored-by: omerfdeniz <omerfaruk.deniz@gmail.com>

* test for duplicate add product to list is added

Co-authored-by: omerfdeniz <omerfaruk.deniz@gmail.com>

omerfdeniz committed:
update

omerfdeniz committed:
update

bekir96 committed:
[BACKEND] B-delete-product-from-list (#410)

* delete product from list method implemented in product_list.py

* test for delete product from list and adding already added product to list is added

* response message typo for invalid product list id is fixed

omerfdeniz committed:
[BACKEND] Implementing the Send Message Functionality (#418)

* initialized the endpoint function

* added /messages to the url paths

* created the Message and Conversation models

* added serializer for the POST requests

* added message class to views/init.py

* added migration files for Message and Conversation models

* implemented the send message functionality

* added message_serializer to serializers/init.py

* added Message and Conversation to models/init.py

* created MessageTest and added setUp() functionality

* added test for sending a message

* added test_message.py to tests/init.py

* changed login check and  added IsAuthenticated permission

* added test for sending message with invalid data

* added test for sending message without login

omerfdeniz committed:
[BACKEND] Implementing Password Change Endpoint (#417)

* implemented the change_password() function

* added /changepassword to the url paths

* changed change_password() permission to IsAuthenticated

* added test for password change

* added test for password change with wrong current password

* added test for password change without logging in

abdullahyldz committed:
mock data and handlers for vendor details

asliayk committed:
[ANDROID] List Dialogs, Guest Options Refactor and Empty List Pages Design (#414)

* purchase options disabled for guest
* accept order with empty shopping cart bug resolved
* visibilities edited, fab borders removed
* list selection alert dialog added
* guest button height increased
* list selection dialogs added
* empty lists page design added
* fab button to listsfragment added
* click listener for fab button added
* empty back accounts design added
* empty address layout added
* vendor page navigation added
* loading states added & visibility changes updated
* gold colored icon created

abdullahyldz committed:
vendor mock handlers

abdullahyldz committed:
vendor refactor delete logs

abdullahyldz committed:
vendor reviews

abdullahyldz committed:
fix mock data for large vendor id

omerfdeniz committed:
[BACKEND] Implementing the Get Conversations Functionality (#422)

* added serializer for messages in the response

* added serializer for conversations in the response

* implemented the get conversations functionality

* added a test for getting all conversations of a logged in user

* added a test for getting all conversations without logging in

MehdiSaffar committed:
Multiple changes

- Make all components work with the form
- Make preview work
- Correctly format images as files
- Correctly format initial values to the form

Co-authored-by: Özdeniz Dolu <ozdeniz.dolu@boun.edu.tr>

omerfdeniz committed:
Delete m2_backup_dump.sql

omerfdeniz committed:
added dump.sql for m2 database backup

MehdiSaffar committed:
fix missing form fields due to currentStep

Co-authored-by: Özdeniz Dolu <ozdeniz.dolu@boun.edu.tr>

MehdiSaffar committed:
process images 64

Co-authored-by: Özdeniz Dolu <ozdeniz.dolu@boun.edu.tr>

MehdiSaffar committed:
format product fields when sending

Co-authored-by: Özdeniz Dolu <ozdeniz.dolu@boun.edu.tr>

MehdiSaffar committed:
only send added/deleted images to backend

Co-authored-by: Özdeniz Dolu <ozdeniz.dolu@boun.edu.tr>

karalleyna committed:
[ANDROID] Vendor Orders & Vendor Order Page (#425)

* Create data models for response and request
* Implement vendor adapter from scratch
* Remove previous adapter
* Design the layout of ViewHolder of VendorOrderAdapter
* Refactor and add order status
* Implement VendorOrderViewModel
* Implement VendorOrdersFragment
* Add two functions sending put and get requests to "vendor/order"
* Add necessary values and icons
* Add dependencies for Google Maps API
* Create VendorOrderFragment
* Add navigation to the single page of the. selected order
* Add shared animationn transition and use Material Design classes
* Create VendorOrderViewModel.kt
* Update design adding necessary views for Material Design components
* Fixes small bug that changes the order of orders
* Implement custom spinner dropdown item
* Create and design fragment_vendor_order
* Hide the API key
* .gitignore refactored
* Revert ".gitignore refactored"
* Revert "Hide the API key"
* key update
* .gitignore updated
* hide api key

furkannane committed:
[Backend] recommendation (#423)

* Recommendation Function is created

Recommendation function is created

* Url is added and init.py import added

Url is added and init.py import added

* tests are added and while adding tests, some corrections are made as well.

tests are added and while adding tests, some corrections are made as well

* Test is updated according to feedback

Test is updated according to feedback: now also checking for status code and is status successful.

* Added status field to the response according to the feedback

Added status field to the response according to the feedback

* Added check for valid user according to the feedback

Added check for valid user according to the feedback

umutoksuz committed:
[BACKEND] Price Change Notifications (#426)

* tests are fixed

* order status change notifications added

* product image update and notification test

* migration added to merge previous migrations

* price change notification added

* notification tests added for price change

* db name in 'settings.py' fixed

* new test added for notifications to test if product is updated but price is not changed

furkan-varol commented on ecommerce/frontend/src/components/VendorContent/VendorSplash.js:
```diff
@@ -0,0 +1,93 @@
+            } catch (error) {
+                console.error(error)
+            } finally {
+                console.log('hi')
```
Unnecessary console.log

furkan-varol reviewed (commented):
I agree with most of the comments made by Mehdi. Solid work overall, thanks for your effort. After the missing functionalities are implemented, I will fully test and approve.

MehdiSaffar committed:
add autocomplete for brand

Co-authored-by: Özdeniz Dolu <ozdeniz.dolu@boun.edu.tr>

umutoksuz committed:
[BACKEND] Add a nonexisting brand when adding product (#433)

* adding a missing brand to database implemented

* a minor bug fixed on edit product

MehdiSaffar committed:
make add product work, add validations, show images in product header

Co-authored-by: Özdeniz Dolu <ozdeniz.dolu@boun.edu.tr>

MehdiSaffar committed:
add suffix to fields, make textareas larger, larger modal

- make discount 0 ==> 100

Co-authored-by: Özdeniz Dolu <ozdeniz.dolu@boun.edu.tr>

MehdiSaffar committed:
ready for merge

Co-authored-by: Özdeniz Dolu <ozdeniz.dolu@boun.edu.tr>

MehdiSaffar committed:
[FRONTEND] Vendor orders (#427)

Co-authored-by: Özdeniz Dolu <ozdeniz.dolu@boun.edu.tr>

ozdenizdolu committed:
Merge branch 'master' into F-product-modal

ozdenizdolu committed:
take care of unnecessary comments

abdullahyldz committed:
merge conflicts with product modal solved

abdullahyldz committed:
add product modal to vendorpage

abdullahyldz committed:
send edit-add product props down to Product Card from vendor

abdullahyldz committed:
modaltype state hook added

abdullahyldz committed:
edit-add product modal correctly visible

abdullahyldz committed:
delete product api call

abdullahyldz committed:
delete product pass props up to parent

abdullahyldz committed:
edit product modal correctly gets product value, added key

abdullahyldz committed:
cleaning

abdullahyldz committed:
min height is given to vendor products page to display spin

abdullahyldz committed:
default edit mode changed to vendor and owner

abdullahyldz committed:
verification mail message update, content.js verification condition update

abdullahyldz committed:
rmv unccessary comment mentioned in PR

abdullahyldz committed:
Merge branch 'master' into F-vendor-page

abdullahyldz committed:
merge conflict with master resolve

ozdenizdolu commented on ecommerce/frontend/src/components/VendorContent/VendorSplash.js:
```diff
@@ -0,0 +1,92 @@
+    }, [])
+
+    const onSendMessage = () => {
+        // Send message modal popup
+        // Sends the message from modal
+        // Go to profile/messages
+        history.push('/profile/messages/')
```
I might be missing something but shouldn't this also make a message modal visible? I couldn't see a message modal anywhere in the code.

MehdiSaffar replied:
I agree with @ozdenizdolu, please check how it is done in Order component

abdullahyldz replied:
I see MessageModal in Purchase.js  
In that file, there is no history.push  like operations. So, I have removed the history.push for now. I didn't get the flow of messaging, should we direct customer to profile/messages after they successfully send a message?

ozdenizdolu commented on ecommerce/frontend/src/components/pages/VendorHomepage.js:
```diff
@@ -0,0 +1,137 @@
+    const onDeleteProductCard = productId => {
+        console.log('delete product card', productId)
+
+        // async function fetch() {
+        //     try {
+        //         const { data } = await api.delete(`/vendor/product`, { id: productId })
+        //     } catch (error) {
+        //         console.error('failed to delete product', error)
+        //     } finally {
+        //     }
+        // }
+
+        // fetch()
```
should this be commented?

abdullahyldz replied:
Commented out, just because I don't have a legit Vendor account with its products. It gets 500 error, I think because of invalid token usage

ozdenizdolu reviewed (commented):
Nicely written. Good job. I will approve after we resolve my comments.

MehdiSaffar commented on ecommerce/frontend/src/components/VendorContent/VendorSplash.js:
```diff
@@ -0,0 +1,92 @@
+        async function fetch() {
+            try {
+                setIsLoading(true)
+                const {
+                    data: { data },
+                } = await api.get(`/vendor/${vendorId}/details`)
```
Real endpoint return the data directly in response, not within data field

abdullahyldz replied:
I fixed it, I will also test it after Backend finalize VendorDetail endpoint.

MehdiSaffar commented on ecommerce/frontend/src/components/VendorContent/VendorSplash.js:
```diff
@@ -0,0 +1,92 @@
+                            <Button
+                                type="primary"
+                                icon={<EditOutlined />}
+                                onClick={() => {
+                                    onSendMessage()
+                                }}>
```
No need to create a function, you can pass `onClick={onSendMessage}` directly

MehdiSaffar commented on ecommerce/frontend/src/components/product_card/ProductCard.js:
```diff
@@ -1,26 +1,59 @@
 import './ProductCard.less'
 import { useAppContext } from '../../context/AppContext'
 import { round, truncate } from '../../utils'
+import { ISO_8601 } from 'moment'
```
Not sure what this is necessary for? I never had to use it personally.

abdullahyldz replied:
It is left from the time you first mentioned about the Date format that Backend sends. I had imported it then, but it is unnecessary now.

MehdiSaffar commented on ecommerce/frontend/src/mocks/handlers.js:
```diff
@@ -27,6 +29,17 @@ const url = u => config.apiUrl + u
+        const minId = min(id, vendorDetails.length - 1) // If request vendorId=55 then get 4 (last element)
+        console.log('returns: ', minId, vendorDetails[minId])
+        return res(
+            ctx.json({
+                data: vendorDetails[minId],
+            }),
```
Return vendorDetails[minId] directly in the response like the backend doe

abdullahyldz replied:
done

MehdiSaffar commented on ecommerce/frontend/src/components/VendorContent/VendorSplash.js:
```diff
@@ -0,0 +1,92 @@
+import { round } from '../../utils'
+import './VendorSplash.less'
+
+const getVendorRatingLevel = ({ rating }) => {
+    if (rating <= 5.0) {
+        return 'low'
+    }
+    if (rating <= 8.0) {
+        return 'medium'
+    }
+    return 'high'
+}
```
In the updated master branch I exported this function from utils. Please use it from there

abdullahyldz replied:
done

MehdiSaffar commented on ecommerce/frontend/src/components/VendorContent/VendorSplash.js:
```diff
@@ -0,0 +1,92 @@
+    return 'high'
+}
+
+export const VendorSplash = ({ vendorId, onEditModeChange, editable }) => {
+    const [vendorHeaderDetails, setVendorHeaderDetails] = useState({})
+    const [isLoading, setIsLoading] = useState(true)
+
+    const { user } = useAppContext()
+    const isVendorAndOwner = user.type === 'vendor' && vendorId === user.id.toString()
+    const history = useHistory()
```
make sure vendorId is passed as an integer from parent component instead (to avoid subtle bugs where you forget it is a string)

abdullahyldz replied:
done

MehdiSaffar reviewed (changes requested):
Good work, needs to be perfected.

abdullahyldz committed:
resolve conflicts

abdullahyldz committed:
message modal added to vendor splash

abdullahyldz committed:
remove iso import, change vendor detail handler

abdullahyldz committed:
vendor rating level from utils

abdullahyldz committed:
delete product finally removal

abdullahyldz committed:
vendor reviews remove unnecessary code

abdullahyldz committed:
remove pagination from vendor reviews

abdullahyldz committed:
vendorId parsed to integer

abdullahyldz committed:
add popconfirm to delete product

abdullahyldz committed:
vendor details backend connection needs image formatting

abdullahyldz committed:
base getflix logo added for undefined vendor image

abdullahyldz committed:
init fix id

abdullahyldz commented:
Added popconfirm to delete product.
Message modal opens and closes. Modal shows the vendor title as receiver.
VendorId is parsed to integer.
Removed pagination tools from Vendor reviews.
Vendor details gets value from Backend. However, Backend is not complete.

MehdiSaffar commented:
@ozdenizdolu and I have the following commments:

- Add a vendor page shortcut (for example in header bar and/or in profile menu)
- It would be better to show edit/delete buttons like in Card or Address
- Either need to implement pagination or increase pageSize to a big number (like 1000), some products are not shown
- Show a spinner in the delete button of a product + refresh the product list so that the product is no longer seen
- Refresh the product list when a product is edited to show changes
- Show notifications for edit/delete successful?
- Allow editing title description profile picture
- Need to divide total_rating / rating_count
- Search products shows product from various vendors, need to add {vendor: vendorId} filter in /search/products
- Products should be sorted by creation/modification date.

abdullahyldz committed:
added typography

MehdiSaffar committed:
Merge branch 'master' into F-vendor-page

MehdiSaffar committed:
multiple changes

- Add a vendor page shortcut (for example in header bar and/or in profile menu)
- It would be better to show edit/delete buttons like in Card or Address
- Either need to implement pagination or increase pageSize to a big number (like 1000), some products are not shown
- Show a spinner in the delete button of a product + refresh the product list so that the product is no longer seen
- Refresh the product list when a product is edited to show changes
- Show notifications for edit/delete successful?
- Allow editing title description profile picture
- Need to divide total_rating / rating_count
- Search products shows product from various vendors, need to add {vendor: vendorId} filter in /search/products
- Products should be sorted by creation/modification date.
- Various other style changes

Co-authored-by: Özdeniz Dolu <ozdeniz.dolu@boun.edu.tr>
Co-authored-by: Abdullah Yıldız <abdullahyildiz321@gmail.com>

MehdiSaffar commented:
We did a long collaborative session over Google meets and vscode Live share and we made all the requested changes as well as polished the vendor home page. We can merge.
